### DISCUSSION TITLE
Troubleshooting a failure to initialize

### DISCUSSION THREAD

**Discussion by prin0315** [Poing-Studios/Godot-AdMob-Android-iOS#88] [Q&A]
Is there a way to print or otherwise log error codes or information on why Admob won't initialize? It seems that _on_AdMob_initialization_complete receives the NOT_READY status, and I can't seem to figure out why or whether its a code issue or an issue with my AdMob account.

**Comment by gumaciel** [comment]
Hi @prin0315 , i tested here to put a description, but if fails the message is this one: "Timeout"

![Screenshot_2021-06-25-21-35-26-403_org.godotengine.admobaddon.jpg](https://user-images.githubusercontent.com/20030153/123496214-51c15900-d5fd-11eb-831a-563510dc0fea.jpg)

I think is still without much detailing, but we can't do anything better than that because we depend of Google and AdMob to do that.

Observation, if the AdMob didn't initialize successfully, then for us it can only be three things:

- Your account is very new and you need to wait for AdMob to activate it, it may take up to 2 weeks
- Your APPLICATION_ID is incorrect
- Your ad IDs are incorrect or they also need to wait a while to activate (but I don't think this causes a startup error, at least it never happened to us)

Read more here: https://support.google.com/admob/answer/9469204?hl=en
https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/wiki/Question-and-Answers#4-why-my-real-ads-are-now-showing
